OpenRA.Game/Server/Server.cs

```diff
@@ -156,11 +156,11 @@ public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData)
 				for (;;)
 				{
 					var checkRead = new List<Socket>();
-					checkRead.Add(listener.Server);
+					if (State == ServerState.WaitingPlayers) checkRead.Add(listener.Server);
 					foreach (var c in Conns) checkRead.Add(c.socket);
 					foreach (var c in PreConns) checkRead.Add(c.socket);
 
-					Socket.Select(checkRead, null, null, timeout);
+					if (checkRead.Count > 0) Socket.Select(checkRead, null, null, timeout);
```

Comment:
This whole file needs a StyleCop cleanup. Waiting for the students at scm.thm.de/openra/openra-develop to merge in theirs.